admin/tests/test_integration.py

```diff
@@ -0,0 +1,413 @@
+ossec_gpg_fpr: 1F544B31C845D698EB31F2FF364F1162D32E7E58
+sasl_domain: gmail.com
+sasl_password: testpassword
+sasl_username: testuser
+securedrop_app_gpg_fingerprint: 1F544B31C845D698EB31F2FF364F1162D32E7E58
+securedrop_app_gpg_public_key: sd_admin_test.pub
+securedrop_app_https_certificate_cert_src: ''
+securedrop_app_https_certificate_chain_src: ''
+securedrop_app_https_certificate_key_src: ''
+securedrop_app_https_on_source_interface: false
+securedrop_supported_locales:
+- de_DE
+- es_ES
+smtp_relay: smtp.gmail.com
+smtp_relay_port: 587
+ssh_users: sd
+'''
+
+HTTPS_OUTPUT = '''app_hostname: app
+app_ip: 10.20.2.2
+daily_reboot_time: 5
+dns_server: 8.8.8.8
+enable_ssh_over_tor: true
+journalist_alert_email: test@gmail.com
+journalist_alert_gpg_public_key: sd_admin_test.pub
+journalist_gpg_fpr: 1F544B31C845D698EB31F2FF364F1162D32E7E58
+monitor_hostname: mon
+monitor_ip: 10.20.3.2
+ossec_alert_email: test@gmail.com
+ossec_alert_gpg_public_key: sd_admin_test.pub
+ossec_gpg_fpr: 1F544B31C845D698EB31F2FF364F1162D32E7E58
+sasl_domain: gmail.com
+sasl_password: testpassword
+sasl_username: testuser
+securedrop_app_gpg_fingerprint: 1F544B31C845D698EB31F2FF364F1162D32E7E58
+securedrop_app_gpg_public_key: sd_admin_test.pub
+securedrop_app_https_certificate_cert_src: sd.crt
+securedrop_app_https_certificate_chain_src: ca.crt
+securedrop_app_https_certificate_key_src: key.asc
+securedrop_app_https_on_source_interface: true
+securedrop_supported_locales:
+- de_DE
+- es_ES
+smtp_relay: smtp.gmail.com
+smtp_relay_port: 587
+ssh_users: sd
+'''
+
+
+def setup_function(function):
+    global SD_DIR
+    SD_DIR = tempfile.mkdtemp()
+    ANSIBLE_BASE = '{0}/install_files/ansible-base'.format(SD_DIR)
+    cmd = 'mkdir -p {0}/group_vars/all'.format(ANSIBLE_BASE).split()
+    subprocess.check_call(cmd)
+    for name in ['sd_admin_test.pub', 'ca.crt', 'sd.crt', 'key.asc']:
+        subprocess.check_call('cp -r {0}/files/{1} {2}'.format(CURRENT_DIR,
+                              name, ANSIBLE_BASE).split())
+    for name in ['de_DE', 'es_ES', 'fr_FR', 'pt_BR']:
+        dircmd = 'mkdir -p {0}/securedrop/translations/{1}'.format(
+            SD_DIR, name)
+        subprocess.check_call(dircmd.split())
+
+
+def teardown_function(function):
+    subprocess.check_call('rm -rf {0}'.format(SD_DIR).split())
+
+
+def verify_username_prompt(child):
+    child.expect("Username for SSH access to the servers:")
+
+
+def verify_reboot_prompt(child):
+    child.expect(
+        "Daily reboot time of the server \(24\-hour clock\):", timeout=2)
+
+
+def verify_ipv4_appserver_prompt(child):
+    child.expect('Local IPv4 address for the Application Server\:', timeout=2)
+
+
+def verify_ipv4_monserver_prompt(child):
+    child.expect('Local IPv4 address for the Monitor Server\:', timeout=2)
+
+
+def verify_hostname_app_prompt(child):
+    child.expect('Hostname for Application Server', timeout=2)
+    # For any default input, the text comes in the following format
```

Comment:
It would help maintenance to have a helper that builds the mangled string from the expected input, **"app".split().join("\x1b[0m")** or something similar. Just an idea ;-)

Reply:
The current form just works with copy paste from the test failures (as the real output in the test), that way we can modify them faster in future. That is why I kept it like this. The upstream also mentioned that those strings may change in future. That is why I tried to keep them as simple as possible.

Reply:
For the record: most of these characters are ANSI escape sequences. Here's me putting in a breakpoint and using a regex I stole from [StackOverflow](https://stackoverflow.com/questions/14693701/how-can-i-remove-the-ansi-escape-sequences-from-a-string-in-python):

```
(Pdb) ansi_escape = re.compile(r'\x1B\[[0-?]*[ -/]*[@-~]')
(Pdb) child.before
' \x1b[0m1\x1b[0m0\x1b[0m.\x1b[0m2\x1b[0m0\x1b[0m.\x1b[0m3\x1b[0m.\x1b[0m2\x1b[52D\x1b[52C\x1b[?7h\x1b[0m\x1b[?12l\x1b[?25h\x1b[?25l\x1b[?7l\x1b[52D\x1b[0m\x1b[J\x1b[0mLocal IPv4 address for the Monitor Server: \x1b[0m1\x1b[0m0\x1b[0m.\x1b[0m2\x1b[0m0\x1b[0m.\x1b[0m3\x1b[0m.\x1b[0m2\x1b[52D\x1b[0m\r\r\n\x1b[J\x1b[?7h\x1b[0m\x1b[?12l\x1b[?25h\x1b[?2004l\x1b[?1l\x1b[6n\x1b[?2004h\x1b[?25l\x1b[0m\x1b[?7l\x1b[0m\x1b[J\x1b[0m'
(Pdb) cleaned_text = child.before
(Pdb) ansi_escape.sub('', cleaned_text).
' 10.20.3.2Local IPv4 address for the Monitor Server: 10.20.3.2\r\r\n'
```

Reply:
so in fb9dd9ac01e7693b34b202b805bfdf942ea72167 I'm just stripping off these characters and then asserting the defaults are present - makes for more readable integration test failures (what future maintainers want), e.g.:

```
    def verify_smtp_port_prompt(child):
        child.expect('SMTP port for sending OSSEC alerts\:', timeout=2)
>       assert ANSI_ESCAPE.sub('', child.buffer) == ' 387'  # Expected default
E       AssertionError: assert ' 587' == ' 387'
E         -  587
E         ?  ^
E         +  387
E         ?  ^
tests/test_integration.py:213: AssertionError
```